File: src/main/java/com/paragon/client/shader/shaders/OutlineShader.java
package com.paragon.client.shader.shaders;

import com.paragon.client.shader.Shader;

import java.awt.*;

import static org.lwjgl.opengl.GL20.*;

public class OutlineShader extends Shader {

    private Color colour = new Color(0, 0, 0);
    private float width = 1F;
    private int fill = 0;
    private int outline = 1;

    public OutlineShader() {
        super("/assets/paragon/glsl/shaders/outline.frag");
    }

    public void setColour(Color colour) {
        this.colour = colour;
    }

    public void setWidth(float width) {
        this.width = width;
    }

    public void setFill(int fill) {
        this.fill = fill;
    }

    public void setOutline(int outline) {
        this.outline = outline;
    }

    @Override
    public void setupUniforms() {
        setupUniform("texture");
        setupUniform("resolution");

        setupUniform("colour");
        setupUniform("width");
        setupUniform("fill");
        setupUniform("outline");
    }

    @Override
    public void updateUniforms() {
        glUniform1i(getUniform("texture"), 0);
        glUniform2f(getUniform("resolution"), 1F / mc.displayWidth, 1F / mc.displayHeight);

        glUniform4f(getUniform("colour"), colour.getRed() / 255f, colour.getGreen() / 255f, colour.getBlue() / 255f, colour.getAlpha() / 255f);
        glUniform1f(getUniform("width"), width);
        glUniform1i(getUniform("fill"), fill);
        glUniform1i(getUniform("outline"), outline);
    }
}

File: src/main/kotlin/com/paragon/client/shader/shaders/OutlineShader.kt
package com.paragon.client.shader.shaders

import com.paragon.client.shader.Shader
import org.lwjgl.opengl.GL20.*
import java.awt.Color

class OutlineShader : Shader("/assets/paragon/glsl/shaders/outline.frag") {

    private var colour = Color(0, 0, 0)
    private var width = 1f
    private var fill = 0
    private var outline = 1

    fun setColour(colour: Color) {
        this.colour = colour
    }

    fun setWidth(width: Float) {
        this.width = width
    }

    fun setFill(fill: Int) {
        this.fill = fill
    }

    fun setOutline(outline: Int) {
        this.outline = outline
    }

    override fun setupUniforms() {
        setupUniform("texture")
        setupUniform("resolution")
        setupUniform("colour")
        setupUniform("width")
        setupUniform("fill")
        setupUniform("outline")
    }

    override fun updateUniforms() {
        glUniform1i(getUniform("texture"), 0)
        glUniform2f(getUniform("resolution"), 1f / minecraft.displayWidth, 1f / minecraft.displayHeight)
        glUniform4f(getUniform("colour"), colour.red / 255f, colour.green / 255f, colour.blue / 255f, colour.alpha / 255f)
        glUniform1f(getUniform("width"), width)
        glUniform1i(getUniform("fill"), fill)
        glUniform1i(getUniform("outline"), outline)
    }

}